pkg/crypto/tinkcrypto/primitive/bbs/api/verifier.go

```diff
@@ -0,0 +1,28 @@
+/*
+Copyright SecureKey Technologies Inc. All Rights Reserved.
+
+SPDX-License-Identifier: Apache-2.0
+*/
+
+package api
+
+// Verifier is the verification interface primitive for BBS+ signatures/proofs used by Tink.
+type Verifier interface {
+	// Verify will verify an aggregated signature of one or more messages against the verifier's public key.
+	// returns:
+	// 		error in case of errors or nil if signature verification was successful
+	Verify(messages [][]byte, signature []byte) error
+
+	// VerifyProof will verify a BBS+ signature proof (generated e.g. by Verifier's DeriveProof() call) with the
+	// verifier's public key.
+	// returns:
+	// 		error in case of errors or nil if signature proof verification was successful
+	VerifyProof(messages [][]byte, proof, nonce []byte) error
+
+	// DeriveProof will create a BBS+ signature proof for a list of revealed messages using BBS signature
+	// (can be built using a Signer's Sign() call) and the verifier's public key.
+	// returns:
+	// 		signature proof in []byte
+	//		error in case of errors
+	DeriveProof(messages [][]byte, signature, nonce []byte, revealedIndexes []int) ([]byte, error)
```

Comment:
DeriveProof doesn't seem to be a Verifier activity.

Reply:
DeriveProof uses the public key of the signer, similarly to Verify and VerifyProof.

You're right about the comment, it's the Signer's public key, not the verifier's. I'll update it.

Reply:
DeriveProof can be moved to the Signer interface, I added to the verifier to interface because it uses the public key, not the private key like Sign().

Let me know if it makes more sense to move it to the Signer interface.

Reply:
I wonder if there should be an additional interface if it doesn't quite fit in either?
@Baha-sk

Reply:
The interface is used (returned) by the key manager's Primitive() call, since we have BBS Signer and Verifier key managers, we only need 2 interfaces types, creating a third type doesn't make sense.